package/MDAnalysis/core/groups.py

```diff
@@ -850,6 +850,115 @@ def center_of_geometry(self, pbc=None, compound='group'):
         return self.center(None, pbc=pbc, compound=compound)
 
     centroid = center_of_geometry
+    
+    @warn_if_not_unique
+    def sum(self, elements, compound='group'):
+        """The sum of elements associated with (compounds of) the group.
+        
+        Computes the sum over the elements of :class:`Atoms<Atom>` in the group.
+        The sum per :class:`Residue`, :class:`Segment`, molecule, or
+        fragment can be obtained by setting the `compound` parameter
+        accordingly.
+
+        Parameters
+        ----------
+        elements : array_like
+            Elements to sum.
+        compound : {'group', 'segments', 'residues', 'molecules', 'fragments'}, optional
+            If ``'group'``, the sum over all elements associated
+            with atoms in the group will
+            be returned as a single value. Else, the sum for all elements
+            of each :class:`Segment`, :class:`Residue`, molecule, or fragment
+            will be returned as an 1d array.
+            Note that, in any case, *only* the positions of :class:`Atoms<Atom>`
+            *belonging to the group* will be taken into account.
+
+        Returns
+        -------
+        sum : numpy.ndarray
+            Sum of the group.
+            If `compound` was set to ``'group'``, the output will be a single
+            value.
+            If `compound` was set to ``'segments'``, ``'residues'``, 
+            ``'molecules'``, or ``'fragments'``, the output will be a 1d array
+            of shape ``(n,)`` where ``n`` is the number of compounds.
+
+        Raises
+        ------
+        ValueError
+            If `compound` is not one of ``'group'``, ``'segments'``,
+            ``'residues'``, ``'molecules'``, or ``'fragments'``.
+        ~MDAnalysis.exceptions.NoDataError
+            If `compound` is ``'molecule'`` but the topology doesn't
+            contain molecule information (molnums) or if `compound` is
+            ``'fragments'`` but the topology doesn't contain bonds.
+
+        Examples
+        --------
+
+        To find the total charge of a given :class:`AtomGroup`::
+
+            >>> sel = u.select_atoms('prop mass > 4.0')
+            >>> sel.sum(sel.charges)
+
+        To find the total mass per residue of all CA :class:`Atoms<Atom>`::
```

Comment:
I'm confused what this will return.  Is it the mass of CA atoms aggregated per residue?  So if each residue only had one CA it would be identical across all?

Reply:
> Is it the mass of CA atoms aggregated per residue?

Yes, accectly.

> So if each residue only had one CA it would be identical across all?

Yes. In the case for only one residue this is a bit senseless. But could be useful if you have for example a geometric selection...